### PR TITLE
[spmd expansion] support scalar_tensor

### PR DESCRIPTION
**Pull request by yifuwang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98233
* #98232
* __->__ #98231
* #98230
* #98229

scalar_tensor is a pure factory function that can't be handled by DTensor prop rule and needs to be currently handled in spmd expansion.

Differential Revision: [D44615039](https://our.internmc.facebook.com/intern/diff/D44615039/)

### CODE REVIEW

**Review by mrshenli — APPROVED** [review]
this one supersedes #98179

File: torch/distributed/_spmd/distribute.py [hunk 1]
```diff
@@ -507,11 +507,24 @@ def _convert_to_distributed(
...
-                node_replacements[node] = replacement
+            if node.target == torch.ops.aten.scalar_tensor.default:
+                node_to_obj[node] = DTensor.from_local(
+                    torch.ops.aten.scalar_tensor(
+                        node.args[0],
+                        dtype=node.kwargs["dtype"],
+                        device=node.kwargs["device"]
+                    ),
+                    schemas[0].mesh,
+                    [Replicate()],
```
Comment (mrshenli) [inline torch/distributed/_spmd/distribute.py:518]: I assume this won't trigger collective in backward pass, as it's `requires_grad=False`?

**Comment by yifuwang** [comment]
@yifuwang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44615039).

**Comment by yifuwang** [comment]
@yifuwang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44615039).

**Comment by yifuwang** [comment]
@yifuwang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44615039).

**Comment by yifuwang** [comment]
Superceded by https://github.com/pytorch/pytorch/pull/98390

**Pull request closed without merging**